src/middlewared/middlewared/client/client.py

```diff
@@ -463,7 +478,7 @@ def from_json(args):
                     if e.error:
                         print(e.error, file=sys.stderr)
                     if e.trace:
-                        print(e.trace['formatted'], file=sys.stderr)
+                        print(e.trace, file=sys.stderr)
```

Comment:
Why did you have to remove formatted here? I suspect this will have side effects.

Reply:
so I put that formatted back in there and purposely made a call without a required field so that there would be a ClientException raised and this is the result:

```
root@surfn9:~ # tmux a
root@surfn9:/mnt/jhol/dataset with space # midclt call -j yes rsync.copy '{"path": "/mnt/jhol/anotherfolder", "remote_host": "localhost", "remote_path": "/mnt/jhol/dataset with space", "mode": "SSH", "direction": "PUSH", "remote_password"Total Progress: [________________________________________] 0.00%
[user] This field is required
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/middlewared/client/client.py", line 468, in main
    rv = c.call(args.method[0], *list(from_json(args.method[1:])), **kwargs)
  File "/usr/local/lib/python3.6/site-packages/middlewared/client/client.py", line 378, in call
    raise ClientException(job['error'], trace=job['exception'])
middlewared.client.client.ClientException: [user] This field is required

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/midclt", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/middlewared/client/client.py", line 482, in main
    sys.exit(1)
  File "/usr/local/lib/python3.6/site-packages/middlewared/client/client.py", line 481, in main
    print(e.trace['formatted'], file=sys.stderr)
TypeError: string indices must be integers
root@surfn9:/mnt/jhol/dataset with space #


```